### PR TITLE
Cleanup dav

### PR DESCRIPTION
**Pull request by CarlSchwan**
- Remove unused class AppEnabledPlugin
- Add more types hinting when possible
- Port to LoggerInterface in some places

Only merge after branching

### CODE REVIEW

**Comment by CarlSchwan** [comment]
> > Remove unused class AppEnabledPlugin
> 
> Added by @icewind1991 in [owncloud/core#9962](https://github.com/owncloud/core/pull/9962), maybe they have opinions on this?

Looking at https://github.com/owncloud-archive/contacts/pull/563/files it seems it is about disabling caldav and the carddav endpoint when the respective apps are disabled. The usage of AppEnabledPlugin has been removed in the two apps since the carddav and caldav backend still work if the calendar and contact apps are disabled or not installed

**Comment by CarlSchwan** [comment]
- [ ] todo fix tests

**Comment by tcitworld** [comment]
Ah yes this is very very old. :spider_web: :broom:

File: apps/dav/lib/CardDAV/PhotoCache.php [hunk 1]
```diff
@@ -40,6 +40,7 @@
 use Sabre\VObject\Parameter;
 use Sabre\VObject\Property\Binary;
 use Sabre\VObject\Reader;
+use Psr\Log\LoggerInterface;
```
Comment (MichaIng) [inline apps/dav/lib/CardDAV/PhotoCache.php:43]: Remove `use OCP\ILogger;` above.

**Review by miaulalala — APPROVED** [review]
Boy those typed properties sure are nice :heart_eyes:

File: apps/dav/lib/Connector/PublicAuth.php [hunk 2]
```diff
@@ -44,28 +44,12 @@
  */
 class PublicAuth extends AbstractBasic {
 	private const BRUTEFORCE_ACTION = 'public_webdav_auth';
+	private IShare $share;
```
Comment (CarlSchwan) [inline apps/dav/lib/Connector/PublicAuth.php:47]:
```suggestion
	private ?IShare $share = null;
```

File: apps/dav/lib/Connector/Sabre/Auth.php [hunk 3]
```diff
@@ -50,33 +50,19 @@
...
-	/** @var string */
-	private $currentUser;
-	/** @var Manager */
-	private $twoFactorManager;
-	/** @var Throttler */
-	private $throttler;
+	private ISession $session;
+	private Session $userSession;
+	private IRequest $request;
+	private string $currentUser;
```
Comment (CarlSchwan) [inline apps/dav/lib/Connector/Sabre/Auth.php:56]:
```suggestion
	private ?string $currentUser = null;
```

File: apps/dav/lib/Connector/Sabre/BlockLegacyClientPlugin.php [hunk 4]
```diff
@@ -36,23 +37,17 @@
  * @package OCA\DAV\Connector\Sabre
  */
 class BlockLegacyClientPlugin extends ServerPlugin {
-	/** @var \Sabre\DAV\Server */
-	protected $server;
-	/** @var IConfig */
-	protected $config;
+	protected Server $server;
```
Comment (CarlSchwan) [inline apps/dav/lib/Connector/Sabre/BlockLegacyClientPlugin.php:40]:
```suggestion
	protected ?Server $server = null;
```

File: apps/dav/lib/Connector/Sabre/ChecksumUpdatePlugin.php [hunk 5]
```diff
@@ -26,14 +26,12 @@
...
 use Sabre\HTTP\RequestInterface;
 use Sabre\HTTP\ResponseInterface;
+use Sabre\DAV\Server;
 
 class ChecksumUpdatePlugin extends ServerPlugin {
-	/**
-	 * @var \Sabre\DAV\Server
-	 */
-	protected $server;
+	protected Server $server;
```
Comment (CarlSchwan) [inline apps/dav/lib/Connector/Sabre/ChecksumUpdatePlugin.php:32]:
```suggestion
	protected ?Server $server = null;
```

File: apps/dav/lib/Connector/Sabre/CommentPropertiesPlugin.php [hunk 6]
```diff
@@ -29,22 +29,17 @@
...
-	protected $server;
-
-	/** @var ICommentsManager */
-	private $commentsManager;
-
-	/** @var IUserSession */
-	private $userSession;
-
-	private $cachedUnreadCount = [];
+	protected Server $server;
```
Comment (CarlSchwan) [inline apps/dav/lib/Connector/Sabre/CommentPropertiesPlugin.php:39]:
```suggestion
	protected ?Server $server = null;
```

File: apps/dav/lib/Connector/Sabre/CopyEtagHeaderPlugin.php [hunk 7]
```diff
@@ -27,16 +27,16 @@
...
 use Sabre\HTTP\ResponseInterface;
+use Sabre\DAV\Server;
 
 /**
  * Copies the "Etag" header to "OC-Etag" after any request.
  * This is a workaround for setups that automatically strip
  * or mangle Etag headers.
  */
 class CopyEtagHeaderPlugin extends \Sabre\DAV\ServerPlugin {
+	private Server $server;
```
Comment (CarlSchwan) [inline apps/dav/lib/Connector/Sabre/CopyEtagHeaderPlugin.php:38]:
```suggestion
	private ?Server $server = null;
```

File: apps/dav/lib/Connector/Sabre/DummyGetResponsePlugin.php [hunk 8]
```diff
@@ -43,8 +44,7 @@
  * @package OCA\DAV\Connector\Sabre
  */
 class DummyGetResponsePlugin extends \Sabre\DAV\ServerPlugin {
-	/** @var \Sabre\DAV\Server */
-	protected $server;
+	protected Server $server;
```
Comment (CarlSchwan) [inline apps/dav/lib/Connector/Sabre/DummyGetResponsePlugin.php:47]:
```suggestion
	protected ?Server $server = null;
```

File: apps/dav/lib/Connector/Sabre/FilesPlugin.php [hunk 9]
```diff
@@ -84,66 +85,28 @@ class FilesPlugin extends ServerPlugin {
...
-	 * @var Tree
-	 */
-	private $tree;
-
-	/**
-	 * @var IUserSession
-	 */
-	private $userSession;
+	/** Reference to main server object */
+	private Server $server;
```
Comment (CarlSchwan) [inline apps/dav/lib/Connector/Sabre/FilesPlugin.php:89]:
```suggestion
	private ?Server $server = null;
```

**Comment by CarlSchwan** [comment]
Also noticed that apps/dav/bin/chunkperf.php is broken (require path is not correct), might be a good idea to remove it completely?

File: apps/dav/lib/BackgroundJob/UploadCleanup.php [hunk 10]
```diff
@@ -79,7 +76,8 @@ protected function run($argument) {
 		// The folder has to be more than a day old
 		$initial = $uploadFolder->getMTime() < $time;
 
-		$expire = array_reduce($files, function (bool $carry, File $file) use ($time) {
+		$expire = array_reduce($files, function (bool $carry, Node $file) use ($time) {
+			/** File $file */
```
Comment (come-nc) [inline apps/dav/lib/BackgroundJob/UploadCleanup.php:80]: This change looks weird, if we know it’s a File, why type it as Node in the parameter list?

File: apps/dav/lib/CalDAV/CachedSubscription.php [hunk 11]
```diff
@@ -51,7 +52,7 @@ public function getPrincipalURI():string {
 	/**
 	 * @return array
 	 */
-	public function getACL():array {
+	public function getACL() {
```
Comment (come-nc) [inline apps/dav/lib/CalDAV/CachedSubscription.php:55]: This is backward, should remove the docblock and keep the return type

Comment (CarlSchwan) [reply apps/dav/lib/CalDAV/CachedSubscription.php:55]: unfortunately, the  \Sabre\CalDAV\Calendar  type does not have modern type hinting, so psalm complains.

File: apps/dav/lib/BackgroundJob/UploadCleanup.php (hunk 10 above)
Comment (CarlSchwan) [reply apps/dav/lib/BackgroundJob/UploadCleanup.php]: I added a type hint for $files to be a `File[]` instead

**Review by come-nc — APPROVED** [review]

**Pull request merged**